packages/block-library/src/html/edit.js

```diff
@@ -27,6 +29,10 @@ export default function HTMLEdit( { attributes, setAttributes, isSelected } ) {
 
 	const instanceId = useInstanceId( HTMLEdit, 'html-edit-desc' );
 
+	const isPreviewMode = useSelect( ( select ) => {
+		return select( blockEditorStore ).getSettings().isPreviewMode;
+	}, [] );
```

Comment:
The `isPreviewMode` is also available via the private key in the [block edit context](https://github.com/WordPress/gutenberg/pull/60544). However, considering the block library code is probably the most copied code in the repo, using a selector makes more sense.